include/swift/Localization/LocalizationFormat.h

```diff
@@ -44,6 +44,8 @@ namespace diag {
 
 using namespace llvm::support;
 
+enum LocalizationProducerState : bool { NotInitialized, Initialized };
```

Comment:
I think this is kinda more descriptive than managing a raw boolean

Reply:
How about we add one more state here - `FailedInitialization`?